### ISSUE TITLE
tailscaled segfaults on startup on NixOS with the "hardened" profile enabled

### ISSUE THREAD

**Issue by danderson** [tailscale/tailscale#928]
The hardened profile turns on a bunch of extra-paranoid security knobs, and I suspect one of them makes Go programs crap themselves. But it does make tailscale unusable on such machines.

**Comment by danderson** [comment]
Only happens when I pull in tailscale 1.2 from the unstable track into a hardened 20.09 system. I suspect the unstable track is building stuff incompatibly somehow.

**Comment by danderson** [comment]
The culprit is the "scudo" memory allocator, a hardened allocator from LLVM. The NixOS hardened profile sets it up as the global memory allocator, replacing the standard glibc one. Something it does causes binaries built in the NixOS unstable track to segfault (I saw both bash and tailscaled crash).

Adjusting severity since I don't expect ~anyone to be in this situation except silly old me, and I expect the problem is more in the interaction of scudo with some other change in the system environment.

**Comment by danderson** [comment]
Upstream bug: https://github.com/NixOS/nixpkgs/issues/100799

tl;dr: something to do with a glibc upgrade, combined with using an LLVM memory allocator, and maybe some accidentally left-over LD_PRELOAD configuration that should have been cleaned up. Either way, definitely not a tailscaled issue.
